applications/pelion_client/README.rst

```diff
@@ -0,0 +1,516 @@
+In such case, enable the external flash support and correctly configure the partition layout.
+
+The Pelion library's update manager is enabled when you select the :option:`CONFIG_PELION_UPDATE` Kconfig option.
+This option enables components required for image transport and storage.
+For the update campaign to be recognized and the image be accepted, the device must be provisioned with the valid update resources.
+To simplify the development process, you can have the update resources created by the update manifest generation tool.
+The generated C file must be stored to the :file:`update_default_resources.c` file, located in the :file:`applications/nrf_desktop/configuration/common` directory.
+The update resources embedded into the application image will be used to provision the device.
+
+Configuration files
+===================
+
+You can find the configuration files in the :file:`applications/pelion_client/configuration` directory.
+For each supported build target, you can find a subdirectory that contains all configuration files for the given target.
+
+The application does not use a single :file:`prj.conf` file.
+Configuration files are provided for different build types for each supported build target.
+
+The following build types are available for various build targets:
+
+* ``ZRelease`` -- Release version of the application with no debugging features.
+* ``ZDebug`` -- Debug version of the application; the same as the ``ZRelease`` build type, but with debug options enabled.
+
+If the given build type is not supported on the selected build target, an error message will appear when `Building and running`_.
+In addition to the build types mentioned above, some build targets can provide more build types, which can be used to generate an application in a specific variant.
+Selecting the build type impacts configuration of all system elements that are enabled (application, bootloader, partition layout).
+
+The following configuration files are specified for each build type:
+
+* :file:`app_${CMAKE_BUILD_TYPE}.conf` - Contains configuration of the application.
+* :file:`mcuboot_${CMAKE_BUILD_TYPE}.conf` - Contains configuration of the bootloader.
+* :file:`pm_static_${CMAKE_BUILD_TYPE}.conf` - Contains configuration of the partition layout.
+
+Additionally, each build target directory contains the following configuration files:
+
+* :file:`mcuboot_private.pem` - Private signature used for binary image signing.
+* :file:`dts.overlay` - The DTS overlay used to adjust the DTS for needs of the application (that is, enable and configure the required hardware).
+* ``_def`` files - Configuration files for application modules.
+
+The :file:`nrf52840dk_nrf52840` directory also includes the following optional file:
+
+* :file:`config-tls-pelion.h` - mbed TLS configuration overlay; used when Zephyr's default mbed TLS version is selected (this is a different mbed TLS version than the one provided by :ref:`nrfxlib:nrf_security`).
```

Comment:
Mbed

Reply:
Checked with mbed.org and in our docs, and it's spelled "mbed TLS".